Update location-retrieval for Spring25

bigludo7 opened the pull request:
#### What type of PR is this?

Add one of the following kinds:

* enhancement/feature



#### What this PR does / why we need it:

Updates location-retrieval API with the changes derived from Commonalities and ICM updates
x-camara-guidelines: 0.5.0


#### Which issue(s) this PR fixes:

<!-- Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`. -->

Fixes #276, #279 

#### Special notes for reviewers:

There are still some open discussions in Commonalities that may impact this PR, so we will keep it open until there is a final version for x-camara-guidelines: 0.5.0

#### Changelog input

```
* Error schemas updated with enums
* Some error status and codes have been updated
* Section with guidelines about how to identify the device from access token and request body has been updated 
*  Test definitions aligned with API specification update

```

#### Additional documentation 

This section can be blank.



```
docs

```

bigludo7 commented on code/API_definitions/location-retrieval.yaml:
```diff
@@ -412,23 +400,33 @@ components:
         application/json:
           schema:
-            $ref: "#/components/schemas/ErrorInfo"
+            allOf:
+              - $ref: "#/components/schemas/ErrorInfo"
+              - type: object
+                properties:
+                  status:
+                    enum:
+                      - 400
+                  code:
+                    enum:
+                      - INVALID_ARGUMENT
+                      - OUT_OF_RANGE
+                      - LOCATION_RETRIEVAL.MAXAGE_INVALID_ARGUMENT
```
Hello @maxl2287 
Humm... Looking for your view because I tried here to strictly follow what we have in commonalities in [CAMARA_common.yaml](https://github.com/camaraproject/Commonalities/blob/main/artifacts/CAMARA_common.yaml)

maxl2287 replied:
Ahh alright.
I just took a look in the [event-subscription-template.yaml for r2.1](https://github.com/camaraproject/Commonalities/blob/r2.1/artifacts/camara-cloudevents/event-subscription-template.yaml)

But I see. 
Maybe we should update the commonalities file for event-subscriptions as well so that nobody is confused, like I was :D

maxl2287 reviewed (approved):
LGTM :)

jlurien commented on code/API_definitions/location-retrieval.yaml:
```diff
@@ -466,23 +471,30 @@ components:
+                  code:
+                    enum:
+                      - PERMISSION_DENIED
+                      - INVALID_TOKEN_CONTEXT
```
We may need to remove INVALID_TOKEN_CONTEXT for our APIs. There is an ongoing discussion in Commonalities

bigludo7 replied:
I tend to prefer to remove it. What is your view @jlurien as we need probably to move forward with this PR? I remove it, we merge this one and we listen the Commonalities version in case we need to reintroduce it?

bigludo7 commented:
@jlurien & @maxl2287 403 INVALID_TOKEN_CONTEXT example removed & Test Case aligned accordingly.

bigludo7 commented:
@jlurien @maxl2287 Aligned with José work in #281 

- Applied the [fix](https://github.com/camaraproject/Commonalities/pull/379)on the note about networkAccessIdentifier support
- Removed 401 AUTHENTICATION_REQUIRED (403 INVALID_TOKEN_CONTEXT was already removed)
- Updated the test plan.

maxl2287 reviewed (dismissed):
Thanks!

maxl2287 commented:
@jlurien can we merge it from your side as well with an approve? :)

jlurien reviewed (commented):
In location-verification PR, I copied from the artifact in Commonalities, the scenarios under 
  # Error scenarios for management of input parameter device

_C01.XX series, and removed the old ones. This way is easier to maintain whatever is changed in the common artifact.

The rest LGTM

jlurien reviewed (approved):
LGTM
